File: python/bubbleSorting.py

"""

Bubble sorting algorithm
By CatProgrammerYT

"""

# Well, I can make a generator of random array, but I`m too lazy to actually
# do that, so... YES
array = [1,23,45,152,612,6,7,132,-213,-12050125,-1]

"""print(array) - modern ways to debug, ya know"""

def bubbleSort(array):
        for i in range(len(array)): # Looping array (will be used as next number then)
                for j in range(len(array)-1): # Looping array (will be used as previous number then)
                        if array[j+1] > array[j]: # If next number is greater than previous
                                array[j], array[j+1] = array[j+1], array[j] # Then change theirs places in array
        return array

sortedArray = bubbleSort(array)

# Debugging tools, ya know
print(sortedArray)

File: go/bubbleSorting.go
/*
	Bubble Sorting algorithm (Golang)
	By CatProgrammerYT
*/

package main

import (
	"encoding/json"
	"fmt"
	"strings"
)

func main() {
	// Random generator will be added later
	var array [11]int = [11]int{1, 2512, 5, 12, 51, 6, 12, 6, 17, 361, 6}
	var sortedArray [len(array)]int = bubbleSorting(array)
	// We are converting int values in string values in array
	s, _ := json.Marshal(sortedArray)

	fmt.Println(strings.Trim(string(s), "[]"))
}

func bubbleSorting(array [11]int) [11]int {
	// i - next number
	for i := 0; i < len(array); i++ {
		// j - previous number
		for j := 0; j < len(array)-1; j++ {
			// if previous is greater than previous
			if array[i] > array[j] {
				// Changing next and previous numbers in array
				array[i], array[j] = array[j], array[i]
			}
		}
	}
	return array
}
